verification/curator-service/api/src/clients/aws-events-client.ts

```diff
@@ -0,0 +1,56 @@
+import AWS from 'aws-sdk';
+
+/**
+ * Client to interact with the AWS CloudWatch Events API.
+ *
+ * This class instantiates the connection to AWS on construction. All
+ * connection configuration (including mocking) should occur on the AWS object
+ * before construction.
+ *
+ * For examples on using the CloudWatch API, see:
+ *   https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/cloudwatch-examples.html
+ */
+export default class AwsEventsClient {
+    private readonly cloudWatchEventsClient: AWS.CloudWatchEvents;
+    constructor(awsRegion: string) {
+        AWS.config.update({ region: awsRegion });
+        this.cloudWatchEventsClient = new AWS.CloudWatchEvents({
+            apiVersion: '2015-10-07',
+        });
+    }
+
+    /**
+     * Proxies a PutRule request to the AWS CloudWatch API.
+     *
+     * For the full API definition, see:
+     *   https://docs.aws.amazon.com/eventbridge/latest/APIReference/API_PutRule.html
+     */
+    putRule = async (
+        ruleName: string,
+        scheduleExpression: string,
+    ): Promise<string> => {
+        const params = {
+            Name: ruleName,
+            ScheduleExpression: scheduleExpression,
+        };
+        try {
+            const response = await this.cloudWatchEventsClient
+                .putRule(params)
+                .promise();
+            this.assertString(response.RuleArn);
+            return response.RuleArn;
+        } catch (err) {
+            console.warn(
+                `Unable to create AWS CloudWatch Events rule with:
+                name: ${ruleName}
+                schedule: ${scheduleExpression}`,
+            );
+            throw err;
+        }
+    };
+
+    private assertString(input: string | undefined): asserts input is string {
+        if (typeof input === 'string') return;
+        else throw new Error('AWS PutRule response missing RuleArn.');
```

Comment:
can any further help be added? What are we supposed to do in that case? when does it happen?

Reply:
It shouldn't happen ever for a successful response, but we need this assertion helper because the field is nullable (/shrug). I debated making this an AssertionError, because that'd be canonical in Java, but that didn't seem conventional in TS.

Reply:
You mean adding an `assert()` call here isn't conventional in TS? how so?

Reply:
ah nm https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions explains that well, lgtm